timeless/access_control/manager_privileges.py

```diff
@@ -0,0 +1,39 @@
+import flask
+
+from timeless.access_control.methods import Method
+from timeless.employees.models import Employee
+
+
+def has_privilege(method=None, resource=None, *args, **kwargs) -> bool:
+    """Check if user with Manager role can access a particular resource."""
+    return __resources.get(resource, lambda *arg: False)(
+        method, *args, **kwargs
+    )
+
+
+def __employee_access(method=None, *args, **kwargs):
+    """
+    @todo #176:30min Add role to the employee model and check that Manager
+     can only access/modify employees that have role of master or
+     interns.
+    """
+    permitted, user = False, flask.g.get("user")
+    employee_id = kwargs.get("employee_id")
+    if user:
+        if employee_id:
+            permitted = check_employee(employee_id, method, user)
+        else:
+            permitted = True
+    return permitted
+
+
+def check_employee(employee_id, method, user):
+    if employee_id == user.id and method == Method.READ:
+        return True
+    else:
+        return user.company_id == Employee.query.get(employee_id).company_id
+
+
+__resources = {
+    "employee": __employee_access,
+}
```

Comment:
@krzyk Why are the methods ``has_priviledge``, ``__employee_access`` and ``check_employee`` duplicated accross ``*_priviledge.py`` files? The ones you added here seem identical to the ones in ``director_priviledges.py``.

Reply:
@amihaiemil because those privileges are in some cases similar (but not exactly, see todos)

Reply:
@amihaiemil Those will be different because they work on different employee roles, right now employee doesn't have role field so those are identical except the todos to add roles